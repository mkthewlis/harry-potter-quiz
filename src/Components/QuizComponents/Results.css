.results__header {
    font-size: 3em;
}

.results__statistics {
    list-style: none;
    font-size: 1.8em;
}

.results__statistics li {
    display: flex;
    align-self: center;
    justify-content: center;
    margin-bottom: 1rem;
}