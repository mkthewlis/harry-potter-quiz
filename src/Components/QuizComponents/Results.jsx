import React from 'react';
import './Results.css';

export const Result = ({result, tenRandomQuestions}) => (
    <> {result > 5 ? (
        <h2 className="results__header">Hogwarts awaits you!</h2>
    ) : (
        <h2 className="results__header">Better luck next time!</h2>
    )}
        <ul className="results__statistics">
            <li>Number of correct answers: {result} of {tenRandomQuestions.length}</li>
            <li>Number of incorrect answers: {tenRandomQuestions.length - result} of {tenRandomQuestions.length}</li>
        </ul>
    </>
);