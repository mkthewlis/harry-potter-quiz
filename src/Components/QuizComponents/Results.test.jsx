import React from 'react';
import { mount } from 'enzyme';
import { Results } from './Results';

describe('Results', () => {
    let wrapper;
    const resultMock = 8;
    const randomQuestionsMock = [
        {
          question: 'What do the initials J.K. Rowling stand for?',
          possibleAnswers: [
            { answer: 'Joanne Kathleen', correctAnswer: true },
            { answer: 'Jane Kathleen', correctAnswer: false },
            { answer: 'Jody Katie', correctAnswer: false },
            { answer: 'Joanne Katie', correctAnswer: false }
          ],
        }
    ];

    beforeEach(() => {
        wrapper = mount(
          <Results result={resultMock} randomQuestions={randomQuestionsMock}/>,
        );
      });
    it('renders without crashing', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should change the title message depending on if results > 5', () => {
        expect(wrapper.find('.results__header').text()).toContain('Hogwarts awaits you!');
    });

    // result is 'out of 1' question as one question object was supplied to test
    it('should show how many answers were correct', () => {
        expect(wrapper.find('.results__statistics').text()).toContain('Number of correct answers: 8 of 1');
    });
});   