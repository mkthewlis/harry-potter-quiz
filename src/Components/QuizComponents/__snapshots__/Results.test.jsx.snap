// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results renders without crashing 1`] = `
<Results
  randomQuestions={
    Array [
      Object {
        "possibleAnswers": Array [
          Object {
            "answer": "Joanne Kathleen",
            "correctAnswer": true,
          },
          Object {
            "answer": "Jane Kathleen",
            "correctAnswer": false,
          },
          Object {
            "answer": "Jody Katie",
            "correctAnswer": false,
          },
          Object {
            "answer": "Joanne Katie",
            "correctAnswer": false,
          },
        ],
        "question": "What do the initials J.K. Rowling stand for?",
      },
    ]
  }
  result={8}
>
  <section>
    <h2
      className="results__header"
    >
      Hogwarts awaits you!
    </h2>
    <ul
      className="results__statistics"
    >
      <li>
        Number of correct answers: 
        8
         of 
        1
      </li>
      <li>
        Number of incorrect answers: 
        -7
         of 
        1
      </li>
    </ul>
  </section>
</Results>
`;
